File: src/automatica.core/Automatica.WebNew/src/app/base/model/user/usergroup2role.ts
import { BaseModel, Model, JsonFieldInfo, JsonProperty } from "../base-model";
import { Role } from "./role";

@Model()
export class UserGroup2Role extends BaseModel {

    @JsonProperty()
    This2Role: string;

    @JsonProperty()
    This2UserGroup: string;

    @JsonProperty()
    This2RoleNavigation: Role;

    public typeInfo(): string {
        return "UserGroup2Role";
    }

    protected getJsonProperty(): Map<string, JsonFieldInfo> {
        return new Map<string, JsonFieldInfo>();
    }
}

File: src/automatica.core/Automatica.Core.Model/Models/User/UserGroup2Role.cs
﻿using System;

namespace Automatica.Core.Model.Models.User
{
   public class UserGroup2Role : TypedObject
    {
        public Guid This2UserGroup { get; set; }
        public Guid This2Role { get; set; }

        public UserGroup This2UserGroupNavigation { get; set; }

        public Role This2RoleNavigation { get; set; }
    }
}
